go/proof.go

```diff
@@ -309,6 +317,40 @@ func getPadding(spec *InnerSpec, branch int32) (minPrefix, maxPrefix, suffix int
 	return
 }
 
+// isPlaceholder returns true if the padding bytes correspond to all empty children
+// on a given side of this branch, ie. it's a valid placeholder for this position
+func isPlaceholder(spec *InnerSpec, branch int32, op *InnerOp, inPrefix bool) bool {
+	idx := getPosition(spec.ChildOrder, branch)
+
+	// compare the prefix bytes with the appropriate number of empty children
+	if inPrefix {
+		prefixCt := len(spec.ChildOrder) - 1 - idx
+		lenDiff := len(op.Prefix) - prefixCt*int(spec.ChildSize)
+		if lenDiff < int(spec.MinPrefixLength) || lenDiff > int(spec.MaxPrefixLength) {
+			return false
+		}
+		for i := 0; i < prefixCt; i++ {
+			// walk back from the end of the prefix, since its length could vary
+			from := len(op.Prefix) - (prefixCt-i)*int(spec.ChildSize)
+			if bytes.Compare(spec.EmptyChild, op.Prefix[from:from+int(spec.ChildSize)]) != 0 {
```

Comment:
Very good use of https://github.com/confio/ics23/blob/53a3a58ab8d224eef792845a65c7e8fc66712b7d/proofs.proto#L185-L186

I realise now I never check this anywhere, and this was a vestige of a spec indended to cover that case, but never implemented.

Reply:
That's what I figured. Glad it was ready to use for this!